developing-key4hep-software/WritingAlgorithms.md

```diff
@@ -1,65 +1,287 @@
 # Writing Gaudi Algorithms
 
+## Gaudi
 
-{% objectives "Learning Objectives" %}
+Gaudi is an event-processing framework. Algorithms can be defined by users and
+Gaudi will take care of running them for each event. In addition, Gaudi has a
+set of services and tools like logging and support for running in a
+multithreaded environment.
 
-This tutorial will teach you how to:
+The relationship between Gaudi with key4hep happens through
+[k4FWCore](https://github.com/key4hep/k4FWCore). k4FWCore has tools and
+utilities needed to be able to use (almost) seamlessly EDM4hep collections in
+Gaudi algorithms. We recommend checking out the
+[tests](https://github.com/key4hep/k4FWCore/tree/main/test/k4FWCoreTest) in this
+repository since they contain examples of algorithms (in particular of
+algorithms using `Gaudi::Functional`).
 
-* write an algorithm for Key4hep
-* interact with the cmake based build system 
-* use other Gaudi components in the algorithms 
+# Gaudi::Functional
+Using `Gaudi::Functional` is the recommended way of creating algorithms. The
+design is simple and at the same time enforces several constraints at
+compile-time, allowing for a quicker development cycle. In particular, we will
+see that our algorithms won't have an internal state and we obtain the benefit
+of being able to run in a multithreaded environment (almost) trivially[^1].
 
-{% endobjectives %}
+[^1]: It's possible to find algorithms written based on GaudiAlg which is going to be removed from future versions of Gaudi. GaudiAlg was substituted by Gaudi::Algorithm, although the recommended way is to use Gaudi::Functional.
 
+## Setup
+We will need Gaudi, k4FWCore and all their dependencies. Installing these by
+ourselves is not easy but there are software stacks on cvmfs, see the
+{doc}`/setup-and-getting-started/README.md` to set up the key4hep stack.
```

Comment:
```suggestion
{doc}`/setup-and-getting-started/README.md` to set up the key4hep stack.
```

Should this be a link?

Reply:
But that's a link, right? At least we have it like that in other parts of the documentation to link to other parts of the documentation

Reply:
I think a link should be `[text](dest)` . At least the current form doesn't render as a link in the github markdown viewer (maybe it does when run through sphinx?)